ssdp.go

```diff
@@ -529,36 +530,42 @@ func (s *Ssdp) inMSearch(st string, req *http.Request, sendTo string) {
 	if st[0] == '"' && st[len(st)-1] == '"' {
 		st = st[1 : len(st)-2]
 	}
-	mx := 1
+	mx := time.Second
 	if mxStr := req.Header.Get("MX"); mxStr != "" {
 		mxInt64, err := strconv.ParseInt(mxStr, 10, 0)
 		if err != nil {
-			mx = int(mxInt64)
+			s.logger.Warnf("Could not parse mxStr: %s", err)
+			return
 		}
-		if mx < 1 {
-			mx = 1
-		} else if mx > 5 {
-			mx = 5
+		mx = time.Duration(mxInt64) * time.Second
+		if mx < time.Second {
+			mx = time.Second
+		} else if mx > 60*time.Second {
+			mx = 60 * time.Second
 		}
 	}
 
 	if st == "ssdp:all" {
+		totalServers := 0
+		for _, v := range s.advertisableServers {
+			totalServers += len(v)
+		}
 		for _, v := range s.advertisableServers {
 			for _, d := range v {
-				s.respondToMSearch(d, sendTo, mx)
+				s.respondToMSearch(d, sendTo, mx/time.Duration(totalServers))
```

Comment:
Maybe we should use a go-routine instead for each server? That way we respond closer to what the client would expect for their requested MX...
I'm guessing you also did it this way to "spread out" the responses? 
Thoughts?

Reply:
I think this way without go routines is better so responses are spread out, as spec says they should be in case of multiple responses.

I also wrote this code to get best results for my application for Amazon Alexa. The way code is written now it detects up to 16 devices on same ip. I'm not sure if that is some inherent limitation of Alexa, but I was unable to get better results. If I change this to go routines only 12 devices get detected. And what is even worse they are detected in random order if go routines are used.